agent/acl_test.go

```diff
@@ -52,11 +36,15 @@ type TestACLAgent struct {
 // Basically it needs a local state for some of the vet* functions, a logger and a delegate.
 // The key is that we are the delegate so we can control the ResolveToken responses
 func NewTestACLAgent(t *testing.T, name string, hcl string, resolveAuthz authzResolver, resolveIdent identResolver) *TestACLAgent {
-	a := &TestACLAgent{Name: name, HCL: hcl, resolveAuthzFn: resolveAuthz, resolveIdentFn: resolveIdent}
-	dataDir := `data_dir = "acl-agent"`
+	a := &TestACLAgent{resolveAuthzFn: resolveAuthz, resolveIdentFn: resolveIdent}
+
+	dataDir := testutil.TempDir(t, "acl-agent")
```

Comment:
I wonder if we should have `testutil.TempDir` call `t.Cleanup()` itself, removing the need for all the callers to add it. Thoughts?

Reply:
I like that! I don't know of anything that would speak against it.